mava/components/jax/training/step.py

```diff
@@ -339,3 +403,18 @@ def config_class() -> Optional[Callable]:
             config class/dataclass for component.
         """
         return MAPGWithTrustRegionStepConfig
+
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        GAE required to set up trainer.store.gae_fn.
+        MAPGEpochUpdate required for config num_epochs and num_minibatches.
+
+        Returns:
+            List of required component classes.
+        """
+        return Step.required_components() + [
```

Comment:
This would be impacted by the above comment, if we opt to change it.

Reply:
Noted, thanks! This is where I am proposing moving `ParallelSequenceAdder` to.
All the other requirements seem necessary for any trainer step one might implement.

Reply:
Can you also please move `ParallelSequenceAdder` here?

Reply:
I agree that parallelsequenceadder should be moved here. Perhaps step could have Adder but maybe that is redundant

Reply:
Done @RuanJohn. Thanks for the feedback!
@AsadJeewa I don't see why an adder would be directly required by Step? If it's required, then it's required implicitly by an upstream dependency.

Reply:
Thanks @mmorris44 :muscle:

Reply:
What I mean is that a concrete extension of step (MAPGWithTrustRegionStep) requires parallelsequenceadder so the abstract super class (Step) would depend on adder